### ISSUE TITLE
[Bug]: [JSON] Create index on JSON field return successfully while it should fail with hint messages

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#24116]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20230515-30415e1b
- Deployment mode(standalone or cluster):both
- MQ type(rocksmq, pulsar or kafka):    all
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.4.0.dev27
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

 Create index on JSON field return successfully

### Expected Behavior

 Fail to create index on JSON field with error messaged

### Steps To Reproduce

```python
from pymilvus import CollectionSchema, FieldSchema
from pymilvus import Collection
from pymilvus import connections
from pymilvus import DataType
from pymilvus import Partition
from pymilvus import utility

dim = 128
int64_field = FieldSchema(name="int64", dtype=DataType.INT64, is_primary=True)
float_field = FieldSchema(name="float", dtype=DataType.FLOAT)
bool_field = FieldSchema(name="bool", dtype=DataType.BOOL)
string_field = FieldSchema(name="string", dtype=DataType.VARCHAR, max_length=65535)
json_field = FieldSchema(name="json_field", dtype=DataType.JSON)
float_vector = FieldSchema(name="float_vector", dtype=DataType.FLOAT_VECTOR, dim=dim)
schema = CollectionSchema(fields=[int64_field, float_field, bool_field, json_field, float_vector])
connections.connect(host="10.102.6.68", port="19530")
collection = Collection("test_search_collection_binbin_tmp_111", schema=schema)

index_param = {"index_type": "IVF_SQ8", "metric_type": "L2", "params": {"nlist": 8192}}

collection.create_index("json_field", index_param, index_name="index_name_1")
Status(code=0, message=)
```

### Milvus Log

It could be reproduced

**Comment by xiaocai2333** [comment]
#24176 has fixed it.
not support indexed on json field yet.
/assign @binbinlv

**Comment by binbinlv** [comment]
@xiaocai2333 
Should this pr be cherry picked to 2.2?

**Comment by binbinlv** [comment]
Verified and fixed in master branch:
milvus: master-20230519-3a66e1de
pymilvus: 2.4.0.dev39

```
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=create index on json field is not supported)>
```

**Comment by binbinlv** [comment]
It still exists in 2.2 branch:

```
>>> index_param = {"index_type": "IVF_SQ8", "metric_type": "L2", "params": {"nlist": 8192}}
>>>
>>> collection.create_index("json_field", index_param, index_name="index_name_1")
Status(code=0, message=)
```

**Comment by binbinlv** [comment]
Verified and fixed in 2.2.* branch:

milvus: [2.2.0-20230520-1eb804fc](https://hub.docker.com/layers/milvusdb/milvus/2.2.0-20230520-1eb804fc/images/sha256-f1fce0b4ba76324119d89cebe860886197da341061e6688f0ef9da34719ef485?context=explore)
pymilvus: 2.2.9.dev12


```
>>> collection.create_index("json_field", index_param, index_name="index_name_1")
RPC error: [create_index], <MilvusException: (code=1, message=create index on json field is not supported)>, <Time:{'RPC start': '2023-05-22 11:24:42.421471', 'RPC error': '2023-05-22 11:24:42.535266'}>
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/binbin/towhee-test-env-3.8-0.9.0/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 959, in create_index
    return conn.create_index(self._name, field_name, index_params, timeout=timeout, **kwargs)
  File "/Users/binbin/towhee-test-env-3.8-0.9.0/lib/python3.8/site-packages/pymilvus/decorators.py", line 109, in handler
    raise e
  File "/Users/binbin/towhee-test-env-3.8-0.9.0/lib/python3.8/site-packages/pymilvus/decorators.py", line 105, in handler
    return func(*args, **kwargs)
  File "/Users/binbin/towhee-test-env-3.8-0.9.0/lib/python3.8/site-packages/pymilvus/decorators.py", line 136, in handler
    ret = func(self, *args, **kwargs)
  File "/Users/binbin/towhee-test-env-3.8-0.9.0/lib/python3.8/site-packages/pymilvus/decorators.py", line 85, in handler
    raise e
  File "/Users/binbin/towhee-test-env-3.8-0.9.0/lib/python3.8/site-packages/pymilvus/decorators.py", line 50, in handler
    return func(self, *args, **kwargs)
  File "/Users/binbin/towhee-test-env-3.8-0.9.0/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 582, in create_index
    raise MilvusException(status.error_code, status.reason)
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=create index on json field is not supported)>
```
